packages/frontend/core/src/components/image-preview/index.tsx

```diff
@@ -16,54 +17,131 @@ import {
 import type { BlockModel, DocCollection } from '@blocksuite/store';
 import clsx from 'clsx';
 import { useErrorBoundary } from 'foxact/use-error-boundary';
-import { useAtom } from 'jotai';
 import type { PropsWithChildren, ReactElement } from 'react';
 import { Suspense, useCallback, useEffect, useRef, useState } from 'react';
-import ReactDOM from 'react-dom';
 import type { FallbackProps } from 'react-error-boundary';
 import { ErrorBoundary } from 'react-error-boundary';
 import useSWR from 'swr';
 
 import { useZoomControls } from './hooks/use-zoom';
-import {
-  captionStyle,
-  cursorStyle,
-  dividerStyle,
-  imageBottomContainerStyle,
-  imagePreviewActionBarStyle,
-  imagePreviewBackgroundStyle,
-  imagePreviewModalCaptionStyle,
-  imagePreviewModalCenterStyle,
-  imagePreviewModalCloseButtonStyle,
-  imagePreviewModalContainerStyle,
-  imagePreviewModalStyle,
-  loaded,
-  scaleIndicatorButtonStyle,
-  unloaded,
-} from './index.css';
-import {
-  hasAnimationPlayedAtom,
-  previewBlockIdAtom,
-  previewblocksAtom,
-} from './index.jotai';
+import * as styles from './index.css';
 
 const filterImageBlock = (block: BlockModel): block is ImageBlockModel => {
   return block.flavour === 'affine:image';
 };
 
+function resolveMimeType(buffer: Uint8Array): string {
+  if (
```

Comment:
any better solution?

Reply:
if I recall correctly @EYHN did something similar somewhere in affine as well in the past. Also this function is brought from the existing code. I extracted for clearance.

Reply:
maybe import `file-type`

Reply:
https://www.npmjs.com/package/@sgtpooki/file-type this one? It seems we already have it as a indirect dependency

Reply:
forget about it, it too big for doing so simple job